app/scripts/directives/pageMessage.dir.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+/*
+  To use the pageMessage function, call the PageMessage Service from the state's controller.
+*/
+
+angular.module('yourStlCourts').directive('pageMessage', function($rootScope){
+  return {
+    restrict: 'E',
+    templateUrl: 'views/pageMessage.html',
+    controller: 'PageMessageCtrl as ctrl',
+    link: function ($scope, element, attrs) {
```

Comment:
You can rid of all this logic. Since you're really just adding and removing a class name, you can use `ng-class` in your directive based on `ctrl.hasMessage()`....and actually you probably don't even need the directive at all. Check this out: https://scotch.io/tutorials/the-many-ways-to-use-ngclass